server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -73,10 +75,14 @@
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
-public class TransportService extends AbstractLifecycleComponent implements ReportingService<TransportInfo>, TransportMessageListener,
-    TransportConnectionListener {
+public class TransportService extends AbstractLifecycleComponent
+        implements ReportingService<TransportInfo>, TransportMessageListener, TransportConnectionListener {
+
     private static final Logger logger = LogManager.getLogger(TransportService.class);
 
+    private static final String PERMIT_HANDSHAKES_FROM_INCOMPATIBLE_BUILDS_KEY = "es.unsafely_permit_handshake_from_incompatible_builds";
+    private static final boolean PERMIT_HANDSHAKES_FROM_INCOMPATIBLE_BUILDS = getPermitHandshakesFromIncompatibleBuilds();
```

Comment:
NIT: maybe just put the logic in this method in a `static { }` block below these two constants to make this easier to read in one go?

Reply:
Ah yeah not sure why I didn't do that. Done in 8132bf33b0b889e84d601a18421fd894e0ed8e36